content/en/blog/2022-06-13-unikraft-releases-hyperion.md

```diff
@@ -0,0 +1,218 @@
+
+`ukstore` has been designed with two methods of registering entries containing getters and setters.
+The first method offers a ways to register folders with entries which can be used in other places in the program and unikernel.
+Each folder is part of a linked list and holds entries also arranged into a linked list.
+This approach is taken to ensure that the implementation does not add new vulnerabilities and so that the attack surface is kept at a minimum.
+This first approach is known as dynamic and occurs at runtime.
+
+However, since runtime registeration can impact the boot-time, another method provided is to use static registration which occurs at compile-time.
+With this method, all entries are created at compile-time but remain to be accessed at runtime, though no new entries can be registered using this method.
+For now, the implemention provided in v0.9.0 only consists of static registration with [on-going working to provide dynamic registration by v0.10.0](https://github.com/unikraft/unikraft/pull/460).
+Overall, when used, both registration methods provide identical functionality, the only difference being that static registrations are not modified as they do not change throughout the execution of the application.
+
+In general the structure of both registration methods is similar.
+The main difference is how the entries and folders are structured in the memory.
+Static entries are ordered in an array saved in a sections table and dynamic entries are held inside folders within memory as an ordered linked list.
+Because of this, searching for static folders is faster, thanks to cache storage locality.
+
+In order to allocate only the space strictly necessary in static registrations, link-time sections are generated for each library that is part of the current build.
+To do this, the build system is invoked to generate a list of all used libraries.
+With this list, headers and linker scripts are generated and are used by the build.
+Each library can easily add new registrations in its own storage and access information stored in others.
+
+#### Usage
+
+To use static registration within your lib `libmylib`, `ukstore` will have to be first included:
+```c
+#include <uk/store.h>
+```
+
+Afterwards you can very simply register your getters & setters via the exposed macro:
+
+```c
+UK_STORE_STATIC_ENTRY(
+  my_registered_entry,
+  type,
+  my_getter,
+  my_setter
+);
+```
+Where:
+- `my_registered_entry` - the name of the ukstore entry (this will be used to access it);
+- `type` - the type of the entry (`s8`, `u8`, `s16`, `u16`, `s32`, `u32`, `s64`, `u64`, `uptr`, `charp`);
+- `my_getter` - the function that returns the metric;
+- `my_setter` - the function that sets a metric.
```

Comment:
```suggestion
- `name` - the name of the ukstore entry (this will be used to access it). Here, it is not a string, so no quoting;
- `type` - the type of the entry (`s8`, `u8`, `s16`, `u16`, `s32`, `u32`, `s64`, `u64`, `uptr`, `charp`);
- `getter` - the reference to the getter function (optional);
- `setter` - the reference to the getter function (optional);
- `cookie` - An extra argument that is handed over to the getter and setter functions;
```